### PR TITLE
Flash refactor and rebalance

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do
The flash no longer has a 2 to 4% chance to burn out on use. Instead it works like this:
- The flash no longer randomly burns out
- If you use it more than 5 times in quick succession it will burn out.
- Every 10 seconds you get one charge back. Using the flash will reset this timer.
- The 5th usage will still work, but it will also burn out the flash

Also, the flash arm implant has a cooldown of 5 seconds instead of 2 seconds.
Also changes synthetic flashes into normal flashes as they were an empty subtype
Also slightly cleans up `flash.dm`

## Why It's Good For The Game
Flashes had a laughably good lifespan, only breaking extremely rarely. This way you actually have to either carry backups or space out your usage, instead of having infinite uses

## Testing
Flashes a skrell until I got the warning message, let it decay to 0 uses.
Then, immediately went to burn it out, slightly spaced out, and noticed that the timer was indeed reset and it took 5 flash uses to burn it out.

## Changelog
:cl:
tweak: Flashes will now guaranteed burn out after using it 5 times in quick succession
tweak: Synthetic flashes have been replaced by normal flashes. This changes nothing.
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/flash.dm [hunk 1]
```diff
@@ -62,43 +63,47 @@
...
-	last_used = world.time
-	times_used = max(0, times_used) //sanity
-
-
 /obj/item/flash/proc/try_use_flash(mob/user)
 	if(broken)
 		return FALSE
 
-	if(cooldown >= world.time && user)
+	if(!COOLDOWN_FINISHED(src, flash_cooldown))
```
Comment (Contrabang) [inline code/game/objects/items/devices/flash.dm:70]:
```suggestion
	if(!COOLDOWN_FINISHED(src, flash_cooldown) && user)
```

File: code/game/objects/items/devices/flash.dm [hunk 2]
```diff
@@ -62,43 +63,47 @@
...
+	addtimer(CALLBACK(src, TYPE_PROC_REF(/atom, set_light), 0), 2 DECISECONDS)
+
 	times_used++
+	if(times_used == (max_uses - 1))
+		to_chat(user, "<span class='warning'>[src] is getting dangerously hot! Don't use it for a few seconds or it will burn out!</span>")
+	else if(times_used == max_uses)
+		burn_out()
 
-	if(user && !clown_check(user))
+	if(HAS_TRAIT(user, TRAIT_CLUMSY) && prob(50))
```
Comment (Contrabang) [inline code/game/objects/items/devices/flash.dm:95]:
```suggestion
	if(user && HAS_TRAIT(user, TRAIT_CLUMSY) && prob(50))
```

**Comment by matttheficus** [comment]
Revs can still infinitely flash, yeah?

**Comment by DGamerL** [comment]
> Revs can still infinitely flash, yeah?

Yup, not changed!

**Comment by dudley500** [comment]
Will supercharging the flash with a powercell  still burn out the flash in one use?

**Comment by DGamerL** [comment]
> Will supercharging the flash with a powercell still burn out the flash in one use?

Not changed

**Review by Sirryan2002 — APPROVED** [review]

File: code/game/objects/items/devices/flash.dm [hunk 3]
```diff
@@ -262,15 +264,14 @@
...
 	name = "memorizer"
-	desc = "If you see this, you're not likely to remember it any time soon."
+	desc = "If you see this, you're not likely to remember it any time soon." // Why doesn't this at least delete your notes smh
 	icon_state = "memorizer"
 	item_state = "nullrod"
 
 /obj/item/flash/armimplant
 	name = "photon projector"
 	desc = "A high-powered photon projector implant normally used for lighting purposes, but also doubles as a flashbulb weapon. Self-repair protocols fix the flashbulb if it ever burns out."
-	cooldown_duration = 2 SECONDS
```
Comment (Contrabang) [inline code/game/objects/items/devices/flash.dm:272]: ERT balance change? 🤨 (add to changelog please)

**Review by Contrabang — DISMISSED** [review]

**Comment by Sirryan2002** [comment]
@DGamerL

File: code/game/objects/items/devices/flash.dm (hunk 3 above)
Comment (DGamerL) [reply code/game/objects/items/devices/flash.dm:272]: Added back - forgot about that!

**Comment by Sirryan2002** [comment]
@Contrabang

**Review by Burzah — APPROVED** [review]

File: code/game/objects/items/devices/flash.dm [hunk 4]
```diff
@@ -62,43 +65,47 @@
...
-
+/obj/item/flash/proc/flash_recharge(mob/user)
+	times_used = max(0, times_used - 1)
+	if(times_used)
+		flash_timer = addtimer(CALLBACK(src, PROC_REF(flash_recharge)), 10 SECONDS, TIMER_STOPPABLE)
+	else
+		flash_timer = null
 
-/obj/item/flash/proc/flash_carbon(mob/living/carbon/M, mob/user = null, power = 10 SECONDS, targeted = 1)
+/obj/item/flash/proc/flash_carbon(mob/living/carbon/M, mob/user, power = 10 SECONDS, targeted = 1)
```
Comment (lewcc) [inline code/game/objects/items/devices/flash.dm:108]: ?
```suggestion
/obj/item/flash/proc/flash_carbon(mob/living/carbon/M, mob/user, power = 10 SECONDS, targeted = TRUE)
```

File: code/game/objects/items/devices/flash.dm [hunk 5]
```diff
@@ -51,8 +54,8 @@
 	battery_panel = !battery_panel
 	return TRUE
 
-/obj/item/flash/random/New()
-	..()
+/obj/item/flash/random/Initialize(mapload)
+	. = ..()
 	if(prob(25))
 		broken = TRUE
 		icon_state = "[initial(icon_state)]burnt"
```
[the remark this answers is not in the document]
Comment (DGamerL) [reply code/game/objects/items/devices/flash.dm:61]: Would that be necessary? `burn_out` does this in the same way

File: code/game/objects/items/devices/flash.dm [hunk 6]
```diff
@@ -13,32 +13,35 @@
...
-				overcharged = TRUE
-				add_overlay("overcharge")
+	if(!can_overcharge || !istype(I, /obj/item/stock_parts/cell))
+		return
+
+	if(battery_panel && !overcharged)
+		to_chat(user, "<span class='notice'>You jam [I] into the battery compartment on [src].</span>")
+		qdel(I)
+		overcharged = TRUE
+		add_overlay("overcharge")
```
[the remark this answers is not in the document]
Comment (DGamerL) [reply code/game/objects/items/devices/flash.dm:44]: Done!

**Pull request merged**